backend/app/controllers/spree/admin/payment_methods_controller.rb

```diff
@@ -49,23 +49,38 @@ def collection
       end
 
       def load_providers
-        @providers = Spree::PaymentMethod.providers.sort_by(&:name)
+        load_payment_methods
       end
+      deprecate load_providers: :load_payment_methods, deprecator: Spree::Deprecation
 
-      def validate_payment_method_provider
+      def load_payment_methods
+        @payment_methods = Rails.application.config.spree.payment_methods.sort_by(&:name)
```

Comment:
I think having a `@payment_methods` ivar can be confusing since, as a developer, I expect to find currently persisted `Spree::PaymentMethod` there instead now it contains possible types of `Spree::PaymentMethod`. What about some more semantic variable name like `@payment_method_types`?